packages/entity-server/src/routes/requests.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { ParamsDictionary, Query } from 'express-serve-static-core';
+import { Request } from 'express';
+import { Context, EntityFields, EntityModel } from '../types';
+
+interface FetchEntityParams extends ParamsDictionary {
+  projectCode: string;
+  entityCode: string;
+}
+
+interface FetchEntityQuery extends Query {
+  fields?: string;
+}
+
+export type EntityResponseObject = {
+  [field in keyof EntityFields]?: EntityFields[field];
+};
+
+// eslint-disable-next-line @typescript-eslint/comma-dangle
+export interface FetchEntityRequest<
+  P = FetchEntityParams,
+  ResBody = EntityResponseObject,
+  ReqBody = Record<string, never>,
+  ReqQuery = FetchEntityQuery
+> extends Request<P, ResBody, ReqBody, ReqQuery> {
+  context: Context<{ entity: EntityModel }>;
```

Comment:
@kael89 This is the solution I found to having the properties of the context object being forced to be optional...
By making `Context` a generic type, each request can specify the context it expects to work with 😄